### PR TITLE
Unban Command

### PR DESCRIPTION
**Pull request by Ikinon**
# Pull request

 - [x] I have read and agreed to the [CODE OF CONDUCT](https://github.com/CascadeBot/CascadeBot/blob/master/CODE_OF_CONDUCT.md).
 - [x] What I'm implementing was assigned to me or was not being worked on and is a somewhat requested feature. For reference see our [trello](https://trello.com/b/gUQehy4l/bot).
 - [x] I have tested all of my changes.
 - [ ] I have followed the style of the rest of the project.
 ## Feature
Unban Command

### CODE REVIEW

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import com.cascadebot.cascadebot.commandmeta.CommandContext;
+import com.cascadebot.cascadebot.commandmeta.ICommandMain;
+import com.cascadebot.cascadebot.commandmeta.Module;
+import com.cascadebot.cascadebot.permissions.CascadePermission;
+import net.dv8tion.jda.core.Permission;
+import net.dv8tion.jda.core.entities.Member;
+import net.dv8tion.jda.core.exceptions.InsufficientPermissionException;
+
+public class UnbanCommand implements ICommandMain {
+
```
Comment (binaryoverload) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java]: Remove blank line

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        if (context.getArgs().length == 0) {
+            context.replyDanger("Not enough arguments (No specified member)");
+            return;
+        }
+        String targetUser = context.getMessage(0);
+        try {
+            context.getGuild().getController().unban(targetUser).queue();
+            context.replyInfo("User <@" + targetUser + "> has been Unbanned!");
```
Comment (binaryoverload) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java]: Why the <@ > ? Just put the name here :)

Comment (binaryoverload) [reply src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java]: You also need to put the message inside the queue success consumer and add an error message

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        } catch (InsufficientPermissionException e) {
+            context.replyWarning("Cannot unban user <@" + targetUser + ">, missing Ban Members permission");
```
Comment (binaryoverload) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java]: Remove the <@>

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        }
+
+    }
+
+
```
Comment (binaryoverload) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java]: Blank line

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+public class UnbanCommand implements ICommandMain {
+
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        if (context.getArgs().length == 0) {
+            context.replyDanger("Not enough arguments (No specified member)");
+            return;
+        }
+        String targetUser = context.getMessage(0);
```
Comment (binaryoverload) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java]: You need to do a format check here. Use `Pattern.compile("[0-9]{17,}")` for now to match it

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+                    banUser(context, bannedUser);
+                });
+            } catch (InsufficientPermissionException e) {
+                context.replyDanger("Cannot get banned user list, missing %s permission!", e.getPermission().getName());
+            }
+        }
+    }
+
+    // TODO: Make a moderation handler that does the action, handling perms etc.
+    private void banUser(CommandContext context, User user) {
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java]: shouldn't this method be `unbanUser`?
```suggestion
    private void unbanUser(CommandContext context, User user) {
```

**Comment by JamesHawkinss** [comment]
[Unban command](https://trello.com/c/plWR1964/24-unban-command)

**Comment by JamesHawkinss** [comment]
[Unban command](https://trello.com/c/plWR1964/24-unban-command)

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        } else {
+            try {
+                context.getGuild().getBanList().queue(list -> {
+                    User bannedUser = list.stream()
+                            .map(Guild.Ban::getUser)
+                            .filter(user -> user.getName().equalsIgnoreCase(target) || user.getAsTag().equalsIgnoreCase(target))
+                            .findFirst()
+                            .orElse(null);
+                    unbanUser(context, bannedUser);
+                });
```
Comment (binaryoverload) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/UnbanCommand.java]: Add a failure consumer

**Pull request merged**